socorro/submitter/breakpad_submitter_utilities.py

```diff
@@ -45,12 +58,20 @@ def save_raw_crash_with_file_dumps(self, raw_crash, dumps, crash_id):
             # Build the payload
             datagen, headers = poster.encode.multipart_encode(raw_crash)
 
+            # requests.post expects a string or buffer, so we run out the datagen
+            # part generator
+            payload = ''.join(datagen)
```

Comment:
I think we can actually remove the `poster` dependency here by using the `files` parameter in requests:

```python
# This variable name is kinda meh
raw_crash_post_files = {key: (None, value) for key, value in raw_crash.iteritems()}
for url in urls:
    resp = requests.post(url, files=raw_crash_post_files)
```

`files` takes a dictionary of data to encode in multipart format. If the values are strings, it assumes the strings are file contents and the key is the name of the file; passing a tuple indicates an explicit filename different from the key, and passing `None` as the filename supresses the output of a filename. I tested this in my terminal:

```python
Python 2.7.11 (default, Feb 25 2016, 12:11:05)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import poster
>>> result = poster.encode.multipart_encode({'foo': 'bar', 'baz': 'biff'})
>>> ''.join(result[0])
'--7ef02afc4e6f4d22815feeb33c455eb5\r\nContent-Disposition: form-data; name="foo"\r\nContent-Type: text/plain; charset=utf-8\r\n\r\nbar\r\n--7ef02afc4e6f4d22815feeb33c455eb5\r\nContent-Disposition: form-data; name="baz"\r\nContent-Type: text/plain; charset=utf-8\r\n\r\nbiff\r\n--7ef02afc4e6f4d22815feeb33c455eb5--\r\n'
>>> import requests
>>> r = requests.post('http://example.com', files={'foo': (None, 'bar'), 'baz': (None, 'biff')})
>>> r.request.body
'--4da50a73f09f47029f71b0ad7ef3bba0\r\nContent-Disposition: form-data; name="foo"\r\n\r\nbar\r\n--4da50a73f09f47029f71b0ad7ef3bba0\r\nContent-Disposition: form-data; name="baz"\r\n\r\nbiff\r\n--4da50a73f09f47029f71b0ad7ef3bba0--\r\n'
```

By default it seems to be missing the `Content-Type` headers for each field that `poster` has. If that's important, you can pass a third parameter in the tuple:

```python
>>> r = requests.post('http://example.com', files={'foo': (None, 'bar', 'text/plain; charset=utf8')})
>>> r.request.body
'--af3641ebfd874dd2b4b78ec1d8feb413\r\nContent-Disposition: form-data; name="foo"\r\nContent-Type: text/plain; charset=utf8\r\n\r\nbar\r\n--af3641ebfd874dd2b4b78ec1d8feb413--\r\n'
```

Reply:
I'm somewhat loathe to remove poster as part of this PR because while I trust you, I have vague memories of having to write a mini_poster in Antenna after hitting issues with poster (didn't support Python 3) and requests `files` argument (I forget what issues I hit--it's entirely possible I just did it wrong).

I think we should push that bit off. I wrote up bug 1411399 (https://bugzilla.mozilla.org/show_bug.cgi?id=1411399) to cover it.